src/Tec/Plugin.php

```diff
@@ -104,26 +139,935 @@ public function register() {
+		$post_type = get_post_type( $post_id );
+
+		// Bail (don't delete) if it's a post type that we don't care about.
+		if ( ! in_array( $post_type, $this->get_supported_post_types( false ), true ) ) {
+			return;
+		}
+
+		/**
+		 * Filter to allow keeping empty metadata.
+		 */
+		$keep_empty_meta = apply_filters( 'tec_labs_wpai_keep_empty_meta', false );
+
+		// Bail if we want to keep empty metadata.
+		if ( $keep_empty_meta ) {
+			$this->add_to_log( "Keeping empty post meta for all." );
+			return;
+		}
+
+		// An array of meta keys that should be preserved even with empty values.
+		$keep_post_meta_meta_keys = [];
+
+		/**
+		 * Allows filtering the meta keys that should be imported even with an empty value.
+		 *
+		 * @var array $keep_post_meta_meta_keys
+		 */
+		$keep_post_meta_meta_keys = apply_filters( 'tec_labs_wpai_keep_post_meta_meta_keys', $keep_post_meta_meta_keys );
+
+		// Bail (don't delete) if we want to keep that empty post meta.
+		if ( in_array( $meta_key, $keep_post_meta_meta_keys, true ) ) {
+			$this->add_to_log( "Keeping empty post meta for `" . $meta_key . "` based on filter." );
+			return;
+		}
+
+		// If the meta value is empty then delete it.
+		if ( empty( $meta_value ) ) {
+			if ( delete_post_meta( $post_id, $meta_key ) ) {
+				$this->add_to_log( "Post meta value for `" . $meta_key . "` was empty and was deleted." );
+			}
+			else {
+				$this->add_to_log( "<span style='color:red;'>Post meta value for " . $meta_key . "was empty BUT post meta could not be deleted.</span>" );
+			}
+		}
+	}
+
+	/**
+	 * Do modifications after a post and its post meta have been saved.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int   $post_id   The ID of the inserted post.
+	 * @param mixed $xml_node  The post data in XML format.
+	 * @param bool  $is_update Whether it's an update or not.
+	 *
+	 * @return void
+	 */
+	function maybe_update_post( int $post_id, mixed $xml_node, bool $is_update ): void {
+		// Convert SimpleXml object to array for easier use.
+		$record = json_decode( json_encode( ( array ) $xml_node ), 1 );
+
+		// Grab the post type of the post being imported.
+		$post_type = get_post_type( $post_id );
+
+		// Check if the post type after the import is still the same.
+		if ( $post_type != $record['posttype'] ) {
+			$this->add_to_log( "<span style='color:red;'><strong>POST TYPES DON'T MATCH!!!</strong></span> Original post type: `" . $record['posttype'] . "`. Post type after import: `" . $post_type . "`." );
+			/**
+			 * Filter to allow keeping a post even if the new post type doesn't match the original one.
+			 */
+			if ( apply_filters( 'tec_labs_wpai_delete_mismatching_post_type', true ) ) {
+				wp_delete_post( $post_id, true );
+				$this->add_to_log( "Post (ID: " . $post_id . ") deleted." );
+			} else {
+				$this->add_to_log( "Post (ID: " . $post_id . ") will be imported based on filter." );
+			}
+		}
+
+		// Update origin for Venues.
+		if ( $post_type === 'tribe_venue' ) {
```

Comment:
@lucatume 
I get your point about the specialized methods.
Here, all post types need to go through the same "transformation" of relinking the linked posts (via `relink_posts()`). I don't see how calling specialized methods per post type would be helpful. It would duplicate the same code.